packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -151,3 +157,319 @@ export class Intc extends Op {
+    hash.digest();
+    stack.push(hash); */
+    // let actualHashedOutput = createKeccakHash('keccak256').update(preImageBytes).digest()
+  }
+}
+
+// SHA512_256 hash of value X, yields [32]byte
+export class Sha512_256 extends Op {
+  execute (stack: TEALStack): void {
+    const hash = sha512_256.create();
+    const val = stack.pop() as Message;
+    hash.update(val);
+    const hashedOutput = Buffer.from(hash.hex(), 'hex');
+    stack.push(hashedOutput);
+  }
+}
+
+// The 32 byte public key is the last element on the stack,
+// preceded by the 64 byte signature at the second-to-last
+// element on the stack, preceded by the data which was signed
+// at the third-to-last element on the stack.
+export class ed25519verify extends Op {
+  execute (stack: TEALStack): void {
+    /* const pubKey = stack.pop();
+    const sig = stack.pop();
+    const data = stack.pop(); */
+    // TO-DO
+  }
+}
+
+export class LessThan extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a < b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class GreaterThan extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a > b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class LessThanEqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a <= b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class GreaterThanEqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a >= b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class And extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a && b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class Or extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a || b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+export class EqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
```

Comment:
```suggestion
    const a = stack.pop();
```
I think this is incorrect. `==` opcode compares `any` with `any`. Not just `bigint` with `bigint`. 
`any` in our case would be: `bigint/Uint8Array` i.e `unit64/byte[]`